tools/clang/test/CodeGenSPIRV/spv.inline.extension.unused.hlsl

```diff
@@ -1,11 +1,21 @@
 // RUN: %dxc -T ps_6_0 -E main -fcgl %s -spirv | FileCheck %s
 
-// CHECK-NOT: OpExtension "some_extension"
-// CHECK-NOT: OpCapability SampleMaskPostDepthCoverage
+// CHECK-NOT: OpExtension "ext_on_type"
+// CHECK-NOT: OpCapability Int8
+// CHECK-NOT: OpExtension "ext_on_field"
+// CHECK-NOT: OpCapability StorageBuffer8BitAccess
 
-[[vk::ext_extension("some_extension")]]
-[[vk::ext_capability(/* SampleMaskPostDepthCoverageCapability */ 4447)]]
-int val;
+// Test that the capabilities and extensions on an unused type are not added
```

Comment:
If we have an unused function using those types, the capabilities/extensions are added no?
(I imagine this might be an issue for inline spirv headers if they add helper functions/classes methods using those annotated types).
(Don't recall if we only compile function we visit starting from the entrypoint or not)

Reply:
We only compile functions that are reachable from the entry point, so it should not be a problem.